src/clr-angular/data/datagrid/providers/column-order-model.service.spec.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ColumnOrderModelService } from './column-order-model.service';
+import { ColumnOrdersCoordinatorService } from './column-orders-coordinator.service';
+import { DatagridHideableColumnModel } from '../datagrid-hideable-column.model';
+import { DomAdapter } from '../../../utils/dom-adapter/dom-adapter';
+import { mockHeaderEl } from './column-order-model.service.mock';
+
+export default function(): void {
+  fdescribe('ColumnOrderCoordinatorService', function() {
+    let columnOrdersCoordinatorService = new ColumnOrdersCoordinatorService();
+    let columnOrderModelService: ColumnOrderModelService;
+    let columnOrderModelServicePrev: ColumnOrderModelService;
+    let columnOrderModelServiceNext: ColumnOrderModelService;
+
+    beforeEach(function() {
+      columnOrdersCoordinatorService = new ColumnOrdersCoordinatorService();
+
+      columnOrderModelService = new ColumnOrderModelService(columnOrdersCoordinatorService, new DomAdapter());
+      columnOrderModelServicePrev = new ColumnOrderModelService(columnOrdersCoordinatorService, new DomAdapter());
+      columnOrderModelServiceNext = new ColumnOrderModelService(columnOrdersCoordinatorService, new DomAdapter());
+
+      // Here visually their columns would appear in the following order:
+      // columnOrderModelServicePrev, columnOrderModelService, columnOrderModelServiceNext;
+      columnOrderModelService.flexOrder = 1;
+      columnOrderModelService.headerEl = mockHeaderEl(200, 40);
+
+      columnOrderModelServicePrev.flexOrder = 0;
+      columnOrderModelServicePrev.headerEl = mockHeaderEl(400, 40);
+
+      columnOrderModelServiceNext.flexOrder = 2;
+      columnOrderModelServiceNext.headerEl = mockHeaderEl(300, 40);
+
+      columnOrdersCoordinatorService.orderModels.push(columnOrderModelService);
+      columnOrdersCoordinatorService.orderModels.push(columnOrderModelServicePrev);
+      columnOrdersCoordinatorService.orderModels.push(columnOrderModelServiceNext);
+    });
+
```

Comment:
You're missing an `afterEach()` to clean up the elements you created with `mockHeaderEl()`.

Reply:
Now cleaning up the elements in `afterEach()`.